File: 118.py
class Solution:
    # @param {integer} numRows
    # @return {integer[][]}
    def generate(self, numRows):
        
        if numRows == 0:
            return []
        
        if numRows == 1:
            return [[1]]
        
        pascal = [[]] * numRows
        pascal[0] = [1]
        pascal[1] = [1,1]
        for k in range(2,numRows):
            temp = [1]
            pre_pascal = pascal[k-1]
            len_pre = len(pre_pascal)
            for i in range(0,len_pre-1):
                temp.append(pre_pascal[i]+pre_pascal[i+1])
            temp.append(1)
            pascal[k] = temp[:]
            
        return pascal
                    
            
        

File: 118.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> mat(numRows);
        for (int i = 0; i < numRows; i++){
            mat[i].resize(i+1);
            mat[i][0] = mat[i][i] = 1;
        }
        
        for(int i =1; i< numRows; i++)
            for(int j = 1; j< i; j++)
                mat[i][j] = mat[i-1][j-1] + mat[i-1][j];
        
        return mat;
            
    }
};